Library/SPIClient.m

```diff
@@ -1380,14 +1379,10 @@ - (void)onSpiConnectionStatusChanged:(SPIConnectionState)newConnectionState {
  */
 - (void)startPeriodicPing {
     __weak __typeof(&*self) weakSelf = self;
-    
-    if (_pingTimer != nil) {
-        [_pingTimer cancel];
-        _pingTimer = nil;
-    }
-    _pingTimer = [[SPIGCDTimer alloc] initWithObject:self queue:"com.assemblypayments.ping"];
-    
-    [_pingTimer afterDelay:0 repeat:true block:^(id self) {
+
+    _pingTimer = [[SPIRepeatingTimer alloc] initWithQueue:"com.assemblypayments.ping"
+                                                 interval:pingFrequency
```

Comment:
the interval should be pingFrequency-pongTimeout... because typically we've already waited (slept) for pongTimeout, except for the first invocation of the loop.

Reply:
This GCD timer won't take into account the run duration of this block in its repeat scheduling, eg it'll start every pingFrequency seconds regardless of how long the block sleeps (unless it takes > pingFrequency, in which case it'll queue up the blocks serially). So this interval in conjunction with me removing the final sleep should mean we get the same effective interval as before.